### ISSUE TITLE
Add ToRefTree instance for Map

### ISSUE THREAD

**Issue by julien-truffaut** [stanch/reftree#4]

**Comment by stanch** [comment]
Currently I don’t have any generic instances, e.g. for sets I have `s.c.i.HashSet` and `s.c.i.TreeSet`. Adding `s.c.i.HashMap` and `s.c.i.TreeMap` should be trivial (especially the latter, since it reuses 99% of the code of `s.c.i.TreeSet`). Now, `Map` points to `s.c.i.Map`, which can be a multitude of things. If the map is created via `Map.apply`, it will always be a `Map1`, `Map2`, `Map3`, `Map4` or a `HashMap`. I can add an instance for `Map` with that assumption, but it will break if some other map is passed in. Likewise, an instance for `Seq` can be added that assumes a `List` (the default for `Seq.apply`), but it would fail on everything else. What do you think?

**Comment by julien-truffaut** [comment]
You're right it makes sense for you to provide diagram for concrete implementations. I am mainly interested by a simple `Map` diagram in order to visualise `Json`

**Comment by stanch** [comment]
In that case what do you think about adding the following?
* `s.c.i.HashMap` and `s.c.i.TreeMap` because that’s low-handing fruit
* `reftree.contrib.SimplifiedInstances.{seq, map}` which would not expose any internal structure, much like https://github.com/stanch/reftree/blob/master/core/src/main/scala/reftree/contrib/SimplifiedInstances.scala#L14

**Comment by stanch** [comment]
I’m open to suggestions w.r.t. the simple map representation. My current idea is to have something like this:
```
             +--------+---------+---------+
             |        |         |         |
             | Map    |    •    |    •    |
             |        |         |         |
             +--------+----X----+-----X---+
                           X         XX
                          XX         X
        XXXXXXXXXXXXXXXXXX           X
    XXXXX                            X
+---X---+-----+-----+             +--X----+-----+-----+
|       |     |     |             |       |     |     |
| Entry | foo | 123 |             | Entry | bar | 456 |
|       |     |     |             |       |     |     |
+-------+-----+-----+             +-------+-----+-----+
```

**Comment by julien-truffaut** [comment]
Good idea 👍

**Comment by stanch** [comment]
Ok, I will add that tonight unless you want to contribute :)

**Comment by julien-truffaut** [comment]
I am not sure I will have the time in the next weeks unfortunately :(

**Comment by stanch** [comment]
![diagram](https://cloud.githubusercontent.com/assets/2670454/19663966/690a5984-9a36-11e6-9111-12f3f2cb0a00.png)

**Comment by stanch** [comment]
Just released `0.7.2` with these changes.

**Comment by julien-truffaut** [comment]
awesome, thank you!
